File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s824230885_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var inputs []string
var inputIndex int

func next() string {
	s := inputs[inputIndex]
	inputIndex++
	return s
}

func hasNext() bool {
	return inputIndex < len(inputs)
}

func nextInt() int {
	s := next()
	n, _ := strconv.Atoi(s)
	return n
}

func nextStrArray() []string {
	s := next()
	return strings.Fields(s)
}

func nextIntArray() []int {
	parts := nextStrArray()
	res := make([]int, len(parts))
	for i, s := range parts {
		res[i], _ = strconv.Atoi(s)
	}
	return res
}

func solve() {
	N := nextInt()
	arr := nextIntArray()
	if N == 0 {
		fmt.Println(0)
		return
	}
	max := arr[0]
	output := 1
	for i := 1; i < N; i++ {
		if arr[i] >= max {
			max = arr[i]
			output++
		}
	}
	fmt.Println(output)
}

func main() {
	inputs = make([]string, 0)
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		inputs = append(inputs, scanner.Text())
	}
	if err := scanner.Err(); err != nil {
		panic(err)
	}
	solve()
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C++/s824230885_comments_removed.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
using namespace std;

vector<string> inputLines;
int currentLine = 0;

string next() {
    if (currentLine < inputLines.size()) {
        return inputLines[currentLine++];
    }
    throw runtime_error("No more input");
}

bool hasNext() {
    return currentLine < inputLines.size();
}

int nextInt() {
    return stoi(next());
}

vector<int> nextIntArray() {
    string line = next();
    stringstream ss(line);
    vector<int> arr;
    int num;
    while (ss >> num) {
        arr.push_back(num);
    }
    return arr;
}

void solve() {
    int N = nextInt();
    vector<int> list = nextIntArray();
    if (N == 0) {
        cout << 0 << endl;
        return;
    }
    int max = list[0];
    int output = 1;
    for (int i = 1; i < N; ++i) {
        if (list[i] > max) {
            max = list[i];
            ++output;
        }
    }
    cout << output << endl;
}

int main() {
    inputLines.clear();
    currentLine = 0;
    string line;
    while (getline(cin, line)) {
        inputLines.push_back(line);
    }
    solve();
    cout.flush();
    return 0; // End of Code
}
